### ISSUE TITLE
Voice recording crashes on Android 10+

### ISSUE THREAD

**Issue by jmartinesp** [vector-im/element-android#6989]
### Steps to reproduce

The reason is unknown at the moment, but an empty file could cause it. I'm guessing it's either a too short voice message which didn't have its data flushed or some kind of path or issue with the logic that converts the uri into the file to read and send it.

It's not related to the Opus library since it only happens on Android versions >= 10, and `VoiceRecorderQ` uses the Opus encoder included in the OS.

#### What did you expect?

The voice recording should work and be sent normally.

#### What happened instead?

It crashes with the error message:

```
Exception: java.lang.RuntimeException: setDataSource failed: status = 0xFFFFFFEA
at android.media.MediaMetadataRetriever.setDataSource(Native Method)
```

### Your phone model

Several

### Operating system version

10+

### Application version and app store

1.4.31, 1.4.32, 1.4.34

### Homeserver

matrix.org

### Will you send logs?

No

### Are you willing to provide a PR?

Yes

**Comment by jmartinesp** [comment]
Can confirm, pressing the voice record button and quickly tapping it again throws the expected exception. The referenced voice message file is empty (size: 0B).

**Comment by jmartinesp** [comment]
It's also preceded by:

```
2022-09-01 17:34:34.505 4262-4320/? W/MediaProvider: Failed to fetch original file descriptor
    java.io.IOException: stat failed: ENOENT (No such file or directory)
        at com.android.providers.media.MediaProvider.getFileFromFileDescriptor(MediaProvider.java:5822)
        at com.android.providers.media.MediaProvider.getOriginalMediaFormatFileDescriptor(MediaProvider.java:5739)
        at com.android.providers.media.MediaProvider.openTypedAssetFileCommon(MediaProvider.java:7162)
        at com.android.providers.media.MediaProvider.openTypedAssetFile(MediaProvider.java:7143)
        at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:548)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:327)
        at android.os.Binder.execTransactInternal(Binder.java:1179)
        at android.os.Binder.execTransact(Binder.java:1143)
     Caused by: android.system.ErrnoException: stat failed: ENOENT (No such file or directory)
        at libcore.io.Linux.stat(Native Method)
        at libcore.io.ForwardingOs.stat(ForwardingOs.java:853)
        at libcore.io.BlockGuardOs.stat(BlockGuardOs.java:420)
        at libcore.io.ForwardingOs.stat(ForwardingOs.java:853)
        at android.app.ActivityThread$AndroidOs.stat(ActivityThread.java:7765)
        at android.system.Os.stat(Os.java:919)
        at com.android.providers.media.MediaProvider.getFileFromFileDescriptor(MediaProvider.java:5816)
        at com.android.providers.media.MediaProvider.getOriginalMediaFormatFileDescriptor(MediaProvider.java:5739) 
        at com.android.providers.media.MediaProvider.openTypedAssetFileCommon(MediaProvider.java:7162) 
        at com.android.providers.media.MediaProvider.openTypedAssetFile(MediaProvider.java:7143) 
        at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:548) 
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:327) 
        at android.os.Binder.execTransactInternal(Binder.java:1179) 
        at android.os.Binder.execTransact(Binder.java:1143) 
2022-09-01 17:34:34.506 1307-27255/? E/MetadataRetrieverClient: offset (0) bigger than file size (0)
```

**Comment by jmartinesp** [comment]
I can also reproduce it some times if the recording gets into an inconsistent state and I lock the device or send the app to background, but I don't really know what causes it. It does look like a race condition of some kind.

**Comment by bmarty** [comment]
Fixed by #7009 I guess.
